src/helpers.rs

```diff
@@ -244,6 +244,13 @@ pub fn get_mango_market_perps_cache(
                 .div(I80F48::from_num(perp_market.base_lot_size))
                 .to_num();
 
+            let root_bank = &mango_group_config.tokens[market_index].root_key;
+            let root_bank = Pubkey::from_str(root_bank.as_str()).unwrap();
+            let node_banks = mango_group_config.tokens[market_index]
+                .node_keys
+                .iter()
+                .map(|x| Pubkey::from_str(x.as_str()).unwrap())
```

Comment:
Unless these are parsed elsewhere, it might not be immediately obvious which key failed to parse, if this line fails.
```suggestion
                .map(|x| match Pubkey::from_str(x.as_str()) {
                    Ok(key) => key,
                    Err(err) => panic!(
                        "Failed to parse 'TODO: where do they come from?' as public key:\n\
                         input: {x}\n\
                         error: {err}"
                    ),
                })
```

Reply:
Not needed we expect that the input file is correctly formatted. Anyways unwrap will panic if it is not.